apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/BigtableClientTest.cs

```diff
@@ -56,6 +57,64 @@ public async Task CheckAndMutateRow_Valid_Request()
+        {
+            var request = new CheckAndMutateRowRequest();
+            Assert.Empty(request.AppProfileId);
+
+            request.AppProfileId = "abc";
+            Assert.Equal("abc", request.AppProfileId);
+
+            request = new CheckAndMutateRowRequest { AppProfileId = "xyz" };
+            Assert.Equal("xyz", request.AppProfileId);
+
+            request.AppProfileId = string.Empty;
+            Assert.Empty(request.AppProfileId);
+        }
+
+        [Fact]
+        public async Task CheckAndMutateRow_AppProfileId_From_Client()
+        {
+            var appProfileId_on_client = "csharp";
+            var appProfileId_on_request = "other";
+            var request = new CheckAndMutateRowRequest
+            {
+                TableNameAsTableName = new TableName("project", "instance", "table"),
+                RowKey = ByteString.CopyFromUtf8("abc"),
+                FalseMutations = {new Mutation()},
+                TrueMutations = {new Mutation()}
+            };
+            var request2 = new CheckAndMutateRowRequest(request);
+            var request_with_appProfileId = new CheckAndMutateRowRequest(request) { AppProfileId = appProfileId_on_request };
+            Mock<BigtableServiceApiClient> mockGrpcClient =
+                new Mock<BigtableServiceApiClient>(MockBehavior.Strict);
+            mockGrpcClient
+                .Setup(x => x.CheckAndMutateRow(request, It.IsAny<CallSettings>()))
+                .Returns(new CheckAndMutateRowResponse());
+            mockGrpcClient
+                .Setup(x => x.CheckAndMutateRow(request_with_appProfileId, It.IsAny<CallSettings>()))
+                .Returns(new CheckAndMutateRowResponse());
+            mockGrpcClient
+                .Setup(x => x.CheckAndMutateRowAsync(request, It.IsAny<CallSettings>()))
+                .ReturnsAsync(new CheckAndMutateRowResponse());
+            mockGrpcClient
+                .Setup(x => x.CheckAndMutateRowAsync(request_with_appProfileId, It.IsAny<CallSettings>()))
+                .ReturnsAsync(new CheckAndMutateRowResponse());
+            BigtableClient client = new BigtableClientImpl(
+                new[] { mockGrpcClient.Object }, appProfileId_on_client);
+            client.CheckAndMutateRow(request);
+            await client.CheckAndMutateRowAsync(request2);
+            Assert.Equal(appProfileId_on_client, request.AppProfileId);
+            Assert.Equal(appProfileId_on_client, request2.AppProfileId);
+            client.CheckAndMutateRow(request_with_appProfileId);
+            Assert.NotEqual(appProfileId_on_client, request_with_appProfileId.AppProfileId);
+            Assert.Equal(appProfileId_on_request, request_with_appProfileId.AppProfileId);
+            await client.CheckAndMutateRowAsync(request_with_appProfileId);
+            Assert.NotEqual(appProfileId_on_client, request_with_appProfileId.AppProfileId);
+            Assert.Equal(appProfileId_on_request, request_with_appProfileId.AppProfileId);
```

Comment:
Sorry for the churn, but I think it would be better to split the tests with the AppProfileId specified on the request into different test methods. They seem to be pretty separated from the other stuff being testing and these are starting to get very cluttered. Also, if you still need local names, please stick with just using the camel casing. For example, `appProfileIdOnRequest` instead of `appProfileId_on_request`. Thanks.